queries/kotlin/highlights.scm

```diff
@@ -27,10 +27,8 @@
 (class_parameter
   (simple_identifier) @variable.member)
 
-(class_body
-  (property_declaration
-    (variable_declaration
-      (simple_identifier) @variable.member)))
+((simple_identifier) @property
+  (#has-parent? @property variable_declaration))
```

Comment:
We need to move this up.

This overrides the query right above it.

Please move it upper

Reply:
This has to be some kind of bug in the tree-sitter query implementation.`#has-parent` does actually very inefficient look up of it's parent. Is the load/query delay also there if you just have (variable_declaration (simple_identifier))?

Reply:
> Is the load/query delay also there if you just have (variable_declaration (simple_identifier))?

Can you clarify this? Do you mean in the original query or the new one?

Reply:
> Is the load/query delay also there if you just have (variable_declaration (simple_identifier))?

From what I tested, yes, it's that pattern causing issue

Reply:
And, yes, that's a parser (most likely) or tree-sitter bug. Nevertheless, it's acceptable as a workaround **iff** there's a comment before it explaining that.

@anthony-S93 can you do that (and also make the corresponding change in `locals.lua`)?

Reply:
@lucario387 Should `@property` be replaced with `@variable.member` as @ObserverOfTime suggested? 

@clason I have added a comment to clarify that the change is only a temporary workaround. Also, do you mean `locals.scm`? Inside `locals.scm`, I see two queries that follow the same pattern in the discussion. 

```txt
(lambda_literal
  (lambda_parameters
    (variable_declaration
      (simple_identifier) @local.definition.parameter)))

(class_body
  (property_declaration
    (variable_declaration
      (simple_identifier) @local.definition.field)))
```

I am not familiar enough with Treesitter's query syntax to confidently make the changes you requested. Should I replace those two queries with something like this?

```txt
((simple_identifier) @local.definition.field
  (#has-parent? @local.definition.field) variable_declaration)
```

Reply:
It's only the `class_body` that's problematic, but since you need to distinguish the two, you probably need both. Your second variant doesn't make any difference since it still has the nesting; the first variant works, though.

@lucario387 ?

Reply:
Idk how other plugins use locals, but pretty sure it will affect all other locals queries.

Maybe we should just comment it out.

Reply:
Yeah, sorry, can't read. The first variant has identical patterns, so the second overwrites the first. You need to check for `lambda_parameters` and `property_declaration` as `#has-ancestor?` (which will be even more expensive, but probably still faster than what we have now).

Reply:
I'd vouch for just commenting it out. `has-ancestor` may have unwanted side-effects

Reply:
Sounds good to me (but just the `class_declaration` one; the `lambda_literal` can stay).

Reply:
> I'd vouch for just commenting it out. `has-ancestor` may have unwanted side-effects

So we comment out both of these?
```txt
(lambda_literal
  (lambda_parameters
    (variable_declaration
      (simple_identifier) @local.definition.parameter)))

(class_body
  (property_declaration
    (variable_declaration
      (simple_identifier) @local.definition.field)))
```

Reply:
> We need to move this up.
> 
> This overrides the query right above it.
> 
> Please move it upper

Just to be clear: the current state of the PR is fine, or does this still need to be done?

Reply:
> So we comment out both of these in locals.scm?

No, just the second one.

Reply:
> Just to be clear: the current state of the PR is fine, or does this still need to be done?

That part is done.